File: lab-1/domain/media_item.py
from abc import ABC, abstractmethod  # Import the ABC (Abstract Base Class) module for defining interfaces


# Define the media_item class as an abstract base class (interface)
class media_item(ABC):
    def __init__(self, title, creation_date):
        self.title = title
        self.creation_date = creation_date

    @abstractmethod
    def play(self):
        pass

    @abstractmethod
    def display_info(self):
        pass

File: src/main/kotlin/media_item/MediaItem.kt
package media_item

import media_item_implementation.MediaItemImplementation
import media_item_implementation.implementations.SongImplementation

abstract class MediaItem: Cloneable {
    abstract val title: String
    abstract val creationDate: String
    abstract val implementation: MediaItemImplementation


    fun play() {
//        getImplementation().play()
        implementation.play()
    }

    abstract fun displayInfo()

}